utam-core/src/main/java/utam/core/framework/base/ElementBuilder.java

```diff
@@ -40,15 +41,29 @@
    * @param <T>    element type
    * @return instance with parameters set in selector
    */
-  public <T extends Actionable> T build(Class<T> type, Object... values) {
+  public <T extends Actionable, R extends BasePageElement> T build(
+      Class<T> type, Class<R> implType, Object... values) {
```

Comment:
we can do `Class implType = Class.forName(type.getName()+"Impl");` and keep method with same parameters

Reply:
Will this work with a generic type specification? We want to ensure that the class created extends `BasePageElement`.

Reply:
I'm not sure what is the benefit of exposing `extends BasePageElement` to be honest. it's implementation details

Reply:
Finding the proper name of the nested implementation class given only the interface class is a challenge. It can be done, I'm sure, but it's a bit of a mess to do so. I'm going to agree that we not make this change for the moment. Perhaps we can revisit the entirety of the `ElementBuilder` code in the future to make it more streamlined.